feature_engine/datetime/datetime.py

```diff
@@ -250,13 +271,32 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         # Check if input data contains same number of columns as dataframe used to fit.
         _check_input_matches_training_df(X, self.n_features_in_)
 
+        # special case index
+        if self.variables_ is None:
+            if self.missing_values == "raise":
+                self._check_index_contains_na(X.index)
+
+            for feat in self.features_to_extract_:
+                X[FEATURES_SUFFIXES[feat][1:]] = FEATURES_FUNCTIONS[feat](
+                    pd.Series(
```

Comment:
I just noticed that with this logic, we would be converting the index to datetime for every feature. 

Could we not convert it one, capture the series outside the loop, and then enter the loop using the already prepared series?

Reply:
Sure. My bad.